feat: adding extra manifests

hamza-m-masood opened the pull request:
## Description

<!-- Provide an overview of what to expect in the PR. -->
<!-- Relate or link the associated epic or task. -->
<!-- Add `@camunda/tech-writers` as reviewer to pull in a tech writer, or add your embedded tech writer. -->
closes: https://github.com/camunda/camunda-platform-helm/issues/2999
## When should this change go live?

<!-- PRs merged go to stage.docs.camunda.io first and must be manually released to docs.camunda.io. -->

- [ ] This is a bug fix, security concern, or something that needs **urgent release support**. (add `bug` or `support` label)
- [ ] This is already available but undocumented and should be released within a week. (add `available & undocumented` label)
- [ ] This is on a **specific schedule** and the assignee will coordinate a release with the DevEx team. (create draft PR and/or add `hold` label)
- [x] This is part of a scheduled **alpha or minor**. (add alpha or minor label)
- [ ] There is **no urgency** with this change (add `low prio` label)

## PR Checklist

<!-- Camunda maintains 18 months of minor versions. Backporting your change to multiple versions is common. -->

- [x] My changes are for **an upcoming minor release** and:
  - [ ] are in the `/docs` directory (version 8.8).
  - [x] are in the `/versioned_docs/version-8.7/` directory (version 8.7).
- [x] My changes are for an **already released minor** and are in a `/versioned_docs` directory.

<!-- UNCOMMENT THIS SECTION IF APPLICABLE. Adding or removing pages requires extra steps.
- [ ] I included my new page in the sidebar file(s).
- [ ] I added a redirect for a renamed or deleted page to the .htaccess file.
-->

- [x] I added a DRI, team, or delegate as a reviewer for technical accuracy and grammar/style:
  - [x] [Engineering team review](https://github.com/camunda/camunda-docs/blob/main/howtos/documentation-guidelines.md#review-process)
  - [x] [Technical writer review](https://github.com/camunda/camunda-docs/blob/main/howtos/documentation-guidelines.md#review-process) via `@camunda/tech-writers` unless working with an embedded writer.

<!-- UNCOMMENT THIS SECTION IF APPLICABLE. Changes to **docs infra**, including updates to workflows and adding new npm packages, must be first discussed via issue or #ask-c8-documentation and linked for context.
- [ ] My changes require a [docs infrastructure review](https://github.com/camunda/camunda-docs/blob/main/howtos/documentation-guidelines.md#review-process). (add `dx` label) -->

hamza-m-masood commented:
@camunda/distribution @conceptualshark this is good to go for an initial review. This can then be backported when the initial review is done.

hamza-m-masood commented:
I can't find an 8.3 label
cc @conceptualshark

hisImminence reviewed (dismissed):
Nice short documentation for this feature! Thanks @hamza-m-masood 

~Only question: this would be for 8.7 and 8.8 docs, right?~

As we just discussed. Fine for me to merge it first and then backport it to all versions in a follow up PR (in alignment with the HCs. 🚀 )

conceptualshark commented on docs/self-managed/setup/guides/add-extra-manifests.md:
```diff
@@ -0,0 +1,39 @@
+---
+id: add-extra-manifests
+title: "Add extra manifests to the Camunda 8 Helm chart"
+sidebar_label: "Add extra manifests"
```
I'm wondering if "Add Kubernetes manifests" would be better in both of these locations. When finding this page in the sidebar, would that be more descriptive (and give an indication that it's deployment-specific)? It might also help with searching for this page.

hamza-m-masood replied:
Changed both to "Add Kubernetes manifests"

conceptualshark commented on docs/self-managed/setup/guides/add-extra-manifests.md:
````diff
@@ -0,0 +1,39 @@
+        test: test
+```
+
+For more information, see the Kubernetes [object documentation](https://kubernetes.io/docs/concepts/overview/working-with-objects/).
````
I wanted to provide a little more information here. Is this the best Kubernetes link? 

I was also wondering if we could give any additional context, like:
- Can you add more than one manifest? What is the format for adding more than one? 
- Does the chart automatically create these resources when deployed? 
- What does maintenance look like for updating these objects? (...This might just be my ignorance here.)

Based on https://github.com/camunda/product-hub/issues/2464, it seems like the use case is deployment customization that doesn't involve modifying the Helm chart - I can call that out specifically, too, with any of the answers to the above.

hamza-m-masood replied:
yes, you can add more manfiests. I can make that clearer by adding a second configmap in the array

> Does the chart automatically create these resources when deployed?
Helm will render the manifest and deploy the rendered manifests to the cluster. I don't think this point is worth mentioning int he guide.

> What does maintenance look like for updating these objects?
The customers maintains the extra manfiests themselves. That's why I listed the points in the best practices section. Also, I don't think this is worth adding to the guide.

hamza-m-masood commented:
@conceptualshark I will assume you have approved since I addressed all your comments. I backported to all relevant versions. Please go ahead and merge.
